addons/http_routing/tests/test_lang_url.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+try:
+    from unittest.mock import patch
+except ImportError:
+    from mock import patch
+from odoo.tests import HttpCase
+
+
+class TestLangUrl(HttpCase):
+    post_install = True
+    at_install = False
+
+    # Mock to simulate hr_HR lang is activated
+    def _get_language_codes(self):
+        return [
+            ('en_US', 'English'),
+            ('hr_HR', 'Croatian / hrvatski jezik'),
+        ]
+
+    # Mock to simulate hr_holidays module is installed
+    def _installed(self):
+        return {
+            module.name: module.id
+            for module in self.env['ir.module.module'].sudo().search(['|', ('state', '=', 'installed'), ('name', '=', 'hr_holidays')])
```

Comment:
or maybe something like `self.env['ir.module.module']._get_installed().update(hr_holidays=42)` to check the real `_get_installed` (and make the test fail if somebody tries to remove it ;-)

Reply:
Yep way better, plus it test the real method and not just copy it in case someone change it at some point, we will still test the real code.

Needed some tweak (storing the original method to avoid recursion) but much cleaner now

Reply:
Shouldn't it be something like `hr_holidays=9999` or it doesn't matter?

Reply:
doesn't matter ;-)